tests/test_api_build.py

```diff
@@ -601,3 +601,13 @@ def test_noarch_foo_value():
     metadata = json.loads(package_has_file(fn, 'info/index.json').decode())
     assert 'noarch' in metadata
     assert metadata['noarch'] == "foo"
+
+
+def test_noarch_python():
+    recipe = os.path.join(metadata_dir, "noarch_python")
```

Comment:
Note that since noarch_python is a folder that is in ```metadata_dir``` and does not start with _, it will be built automatically as part of the test_recipe_builds test, which collects all such folders.  You probably want to rename your noarch_python folder to start with an underscore, and then also make this particular test be an xfail right now.

You might also want to break it up into more tests, so that you can only have smaller parts xfail (the parts you have commented now), while you test known-good parts in other tests.